File: hw/caizapanta/u3/HW18AbstractFactory/src/ec/edu/espe/model/WinFactory.java
package ec.edu.espe.model;

/**
 * 
 * @author Tammy Caizapanta
 */

public class WinFactory extends GUIFactory {
    
    @Override
    public Button createButton() {
        return new WinButton();
    }

    @Override
    public Menu createMenu() {
        return new WinMenu();
    }
    
}

File: hw/vaca/u3/AbstractFactoryJs/WinFactory.js
class WinFactory extends GUIFactory {
    createButton() {
        return new WindowsButton();
    }
    
    createMenu() {
        return new WindowsMenu();
    }
}

// js/factories/LinuxFactory.js
class LinuxFactory extends GUIFactory {
    createButton() {
        return new LinuxButton();
    }
    
    createMenu() {
        return new LinuxMenu();
    }
}